platform/core/src/io/file_system.rs

```diff
@@ -1,16 +1,140 @@
 use crate::core_failure::CoreFailure;
-use std::fs;
+use std::fs::{self, DirBuilder, File};
+use std::path::Path;
+use url::Url;
+use zip::ZipArchive;
 
 pub trait FileSystem {
-    fn read_contents_of_file(&self, path: &str) -> Result<Vec<u8>, CoreFailure>;
+    fn create_dir_structure(&self, path: &str) -> Result<(), CoreFailure>;
```

Comment:
What I find missing from here is anything related to the feature that creates it, which will be eventually related to the permission system. Probably Rust has a very fine-tuned system for the "ownership" of variables and so on. Can we use it here?

Reply:
That should not be in the FileSystem but in the FeatureFileSystem. They are different. 

The FeatureFileSystem creates the path, and it knows about the feature. FileSystem is just a dumb slave.